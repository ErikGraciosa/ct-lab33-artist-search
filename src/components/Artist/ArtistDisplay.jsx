import React from 'react'
import ArtistItem from './ArtistItem'

const ArtistDisplay = ({ artists }) => {


    return (
        <ul>
            {artists.map(artist => (
                <li key={artist.id}>
                    <ArtistItem {...artist} />
                </li>
            ))}

        </ul>
    )
}
export default ArtistDisplay;
