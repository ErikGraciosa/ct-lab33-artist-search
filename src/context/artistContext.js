import React, { createContext, useState, useContext } from 'react';

export const ArtistContext = createContext(null)

export const ArtistProvider = ({ children }) => {
    const [artist, setArtist] = useState('')
    return (
        <ArtistContext.Provider value={{ artist, setArtist }}>
            {children}
        </ ArtistContext.Provider>
    )
}

export const useArtist = () => {
    const { artist } = useContext(ArtistContext)
    const { setArtist } = useContext(ArtistContext)
    return {
        artist,
        setArtist
    }
}
