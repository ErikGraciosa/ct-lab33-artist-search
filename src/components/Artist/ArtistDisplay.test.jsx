import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArtistDisplay from './ArtistDisplay';
import { MemoryRouter } from 'react-router-dom';
import { ArtistProvider } from '../../context/artistContext';

describe('ArtistDisplay component', () => {

  const artists = [{
    id: '1234'
  }];

  afterEach(() => cleanup());
  it.only('renders ArtistDisplay', () => {
    const { asFragment } = render(
      <ArtistProvider>
        <MemoryRouter>
          <ArtistDisplay artists={artists} {...artists} />
        </MemoryRouter>
      </ArtistProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
