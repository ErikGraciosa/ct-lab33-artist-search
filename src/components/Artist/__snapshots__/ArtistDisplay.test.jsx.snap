// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistDisplay component renders ArtistDisplay 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/artist/1234"
      >
         
        <div />
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
